tools/clang/lib/CodeGen/CGExpr.cpp

```diff
@@ -1757,6 +1759,36 @@ void CodeGenFunction::EmitStoreThroughBitfieldLValue(RValue Src, LValue Dst,
   }
 }
 
+// HLSL Change - begin
+static bool IsHLSubscriptOfTypedBuffer(llvm::Value *V) {
+  llvm::CallInst *CI = nullptr;
+  llvm::Function *F = nullptr;
+
+  if ((CI = dyn_cast<llvm::CallInst>(V)) &&
+    (F = CI->getCalledFunction()) &&
+    hlsl::GetHLOpcodeGroup(F) == hlsl::HLOpcodeGroup::HLSubscript)
+  {
+    for (llvm::Value *arg : CI->arg_operands()) {
+      llvm::Type *Ty = arg->getType();
+      if (Ty->isPointerTy()) {
+        std::tuple<bool, hlsl::DXIL::ResourceClass, hlsl::DXIL::ResourceKind> Result =
+          hlsl::dxilutil::GetHLSLResourceType(Ty->getPointerElementType());
+
+        if (std::get<0>(Result) && 
+          std::get<1>(Result) == hlsl::DXIL::ResourceClass::UAV &&
+          // These are the types of buffers that are not typed
+          std::get<2>(Result) != hlsl::DXIL::ResourceKind::StructuredBuffer &&
```

Comment:
Could we be more precise here with a whitelist to only bring in the ones we explicitly want?

Something like
`DxilResource::IsAnyTexture(Kind) || Kind == TBuffer || Kind == TypedBuffer`

(not sure if we want both TBuffer and TypedBuffer here?)

Reply:
Fixed. I don't think TBuffer can be written to.